### ISSUE TITLE
Is it possible to control/add an animation when changing the theme at runtime?

### ISSUE THREAD

**Issue by jorgecoca** [flutter/flutter#20463]
I have created a Flutter application in which I change the theme using the BLOC pattern, like this:

```
import 'package:flutter/material.dart';
import 'package:connected_poc/home_page.dart';
import 'package:connected_poc/theme/theme_switch_bloc.dart';
import 'vehicle_list_page.dart';
import 'theme_page.dart';

void main() => runApp(new MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  ThemeSwitchBloc _themeSwitchBloc;

  @override
  void initState() {
    super.initState();
    _themeSwitchBloc = ThemeSwitchBloc();
  }

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<ThemeData>(
      stream: _themeSwitchBloc.themeDataStream,
      initialData: _themeSwitchBloc.initialTheme,
      builder: (BuildContext context, AsyncSnapshot<ThemeData> snapshot) {
        return MaterialApp(
          initialRoute: null,
          routes: {
            '/': (context) => HomePage(bloc: _themeSwitchBloc),
            '/theme_selector': (context) => ThemePage(bloc: _themeSwitchBloc),
          },
          onGenerateRoute: (RouteSettings settings) =>
              MaterialPageRoute(builder: (context) => Text('booo')),
          title: 'Connected',
          theme: snapshot.data,
        );
      },
    );
  }
}
```

This works fine; `ThemeSwitchBloc` emits a new `ThemeData` and the UI refreshes; however, I would like to control the transition to a new theme, let's say by slowing down the existing animation. Any ideas? Thanks!

**Comment by jorgecoca** [comment]
TL;DR; Yes, you can do this. There's already a class called `ThemeDataTween` that can be consumed, as any other animation, to achieve this goal.
